src/WooCommerce.php

```diff
@@ -390,6 +395,11 @@ public static function getProductDetailsHtmlDataAttr(\WC_Product $product, $is_d
           $selected_attributes[] = $attribute_term_data ? $attribute_term_data->name : $attribute;
         }
       }
+      if (!$selected_attributes && 'variation' === $product->get_type()) {
+        foreach ($attributes as $attribute_name => $options) {
+          $selected_attributes[] = $options;
```

Comment:
This direct assignment of options seems to suggest that the preceding foreach loop will not actually do something for variations, as it only applies to variable products (i.e. the parent products).  Did you test that?  If that is the case, then the preceding foreach should only be executed for variable products (by wrapping it into a condition just like in the new block here).

Reply:
I think that's what I had in mind originally but in a code review there was an attempt to compress it.

Reply:
Could my latest commit solve this?

Reply:
This seems a bit slow checking the product type on every loop.